### PR TITLE
Restore original order of agent startup functions

### PR DESCRIPTION
**Pull request by dereknola**
#### Proposed Changes ####
- Partial revert of #9184, restoring the orginal order of setting up container runtimes, closing the channel, then running the kubelet.
- Renamed `AgentReady` to `ContainerRuntimeReady` to better reflect what the channel is actually tracking.
- Closed E2E coverage gap around this issue by increasing number of etcd nodes in the split role test.

#### Types of Changes ####
Bugfix

#### Verification ####
Manually running split servers (3 etcd, 2 cp, 1 agent), the cluster now initialized all etcd nodes correctly. 

#### Testing ####
Change is now covered by E2E test, and the docker/drone `test` stage now covers this case.

#### Linked Issues ####
https://github.com/k3s-io/k3s/issues/9540

#### User-Facing Change ####

```release-note

```

#### Further Comments ####
I will not be squashing this, as we want the revert commit to stand alone from the other 2 "new" changes.

### CODE REVIEW

File: pkg/agent/run.go [hunk 1]
```diff
@@ -130,19 +131,31 @@ func run(ctx context.Context, cfg cmds.Agent, proxy proxy.Proxy) error {
...
+	} else if nodeConfig.ContainerRuntimeEndpoint == "" {
+		if err := containerd.SetupContainerdConfig(nodeConfig); err != nil {
+			return err
+		}
+		if err := executor.Containerd(ctx, nodeConfig); err != nil {
+			return err
+		}
 	}
-
 	// the agent runtime is ready to host workloads when containerd is up and the airgap
```
Comment (brandond) [inline pkg/agent/run.go:146]:
```suggestion
	// the container runtime is ready to host workloads when containerd is up and the airgap
```

File: pkg/cli/server/server.go [hunk 2]
```diff
@@ -513,7 +513,7 @@ func run(app *cli.Context, cfg *cmds.Server, leaderControllers server.CustomCont
 	}
 
 	agentConfig := cmds.AgentConfig
-	agentConfig.AgentReady = agentReady
+	agentConfig.ContainerRuntimeReady = agentReady
```
Comment (brandond) [inline pkg/cli/server/server.go:516]:
```suggestion
	agentConfig.ContainerRuntimeReady = containerRuntimeReady
```

File: pkg/agent/run.go (hunk 1 above)
Comment (brandond) [reply pkg/agent/run.go:146]: And also at https://github.com/k3s-io/k3s/blob/1c1746114c94498f8b77e0ba13c43ebe6e0da56a/pkg/cli/server/server.go#L113-L117

File: pkg/cli/server/server.go (hunk 2 above)
Comment (brandond) [reply pkg/cli/server/server.go:516]: And also at 
* https://github.com/k3s-io/k3s/blob/1c1746114c94498f8b77e0ba13c43ebe6e0da56a/pkg/cli/server/server.go#L113-L117
* https://github.com/k3s-io/k3s/blob/1c1746114c94498f8b77e0ba13c43ebe6e0da56a/pkg/daemons/config/types.go#L364-L370

**Review by brandond — CHANGES_REQUESTED** [review]

File: pkg/etcd/etcd.go [hunk 3]
```diff
@@ -451,7 +451,7 @@ func (e *ETCD) Start(ctx context.Context, clientAccessInfo *clientaccess.Info) e
 			select {
 			case <-time.After(30 * time.Second):
 				logrus.Infof("Waiting for agent to become ready before joining etcd cluster")
```
Comment (brandond) [inline pkg/etcd/etcd.go:453]: should we do this too?
```suggestion
				logrus.Infof("Waiting for container runtime to become ready before joining etcd cluster")
```

**Review by brandond — APPROVED** [review]

**Review by HarrisonWAffel — APPROVED** [review]

**Review by cwayne18 — APPROVED** [review]

**Pull request merged**